src/main/java/com/uwaterloo/connect/repository/TokenRepository.java

```diff
@@ -0,0 +1,25 @@
+package com.uwaterloo.connect.repository;
+
+
+import com.uwaterloo.connect.model.Token;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Modifying;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.time.LocalDateTime;
+import java.util.Optional;
+
+
+@Repository
+@Transactional(readOnly = true)
+public interface TokenRepository extends JpaRepository<Token, Long> {
+
+    Optional<Token> findByToken(String token);
+
+    @Transactional
+    @Modifying
+    @Query("UPDATE Token c " + "SET c.confirmedAt = ?2 " + "WHERE c.token = ?1")
```

Comment:
@JobanThind , Would it be better to use hibernate APIs rather than a hard coded query?

Reply:
This is more performant.